### PR TITLE
Add-Schema-UserManagement

### PR DESCRIPTION
**Pull request by KustoKing**
Required items, please complete
   
   Change(s):
   - Adding User Management schema

   Reason for Change(s):
   - New schema

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @KustoKing, we will provide you with an update on this soon, but in the meantime, can you please resolve branch conflicts by updating master in your branch?

**Comment by v-atulyadav** [comment]
Hi @KustoKing, can you please resolve branch conflicts by updating master in your branch. Thanks

**Comment by KustoKing** [comment]
> Hi @KustoKing, can you please resolve branch conflicts by updating master in your branch. Thanks

Done

**Comment by KustoKing** [comment]
> Hi @KustoKing, we will provide you with an update on this soon, but in the meantime, can you please resolve branch conflicts by updating master in your branch?

Done

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
thanks for resolving conflicts we will provide you with an update on this soon.
Thanks

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
We will review the Pull Request internally and get back to you by shortly.
Thanks

File: ASIM/schemas/ASimUserManagement.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+Schema:
```
Comment (vakohl) [inline ASIM/schemas/ASimUserManagement.yaml:1]: Is this included somewhere?
image

**Review by vakohl — CHANGES_REQUESTED** [review]

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 2]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
 TargetOriginalUserType,string,Optional,ProcessEvent,,,
 TargetScopeId,string,Optional,ProcessEvent,,,
 TargetScope,string,Optional,ProcessEvent,,,
+ActingAppId,string,Optional,UserManagement,,,
+ActingAppType,string,Optional,UserManagement,Enumerated,Process|Browser|Service|Resource|URL|SaaS application|Other,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1124]: From this CSV (Line 6,8 and 454). Ofer added it back in 19 Dec 2022.

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 3]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+ActiveAppName,string,Optional,UserManagement,,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1125]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 4]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventProduct,string,Mandatory,UserManagement,,,
+EventProductVersion,string,Optional,UserManagement,,,
+EventReportUrl,string,Optional,UserManagement,,,
+EventResult,string,Mandatory,UserManagement,Enumerated,,
+EventResult,string,Mandatory,UserManagement,Enumerated,Success|Partial|Failure|NA,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,NotAuthorized|Other,
+EventSchema,string,Mandatory,UserManagement,,,
+EventSchema,string,Mandatory,UserManagement,Enumerated,UserManagement,
+EventSchemaVersion,string,Mandatory,UserManagement,,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1170]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 5]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventOriginalUid,string,Optional,UserManagement,,,
+EventOwner,string,Optional,UserManagement,,,
+EventProduct,string,Mandatory,UserManagement,,,
+EventProductVersion,string,Optional,UserManagement,,,
+EventReportUrl,string,Optional,UserManagement,,,
+EventResult,string,Mandatory,UserManagement,Enumerated,,
+EventResult,string,Mandatory,UserManagement,Enumerated,Success|Partial|Failure|NA,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,NotAuthorized|Other,
+EventSchema,string,Mandatory,UserManagement,,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1168]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 6]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
 TargetScope,string,Optional,ProcessEvent,,,
+ActingAppId,string,Optional,UserManagement,,,
+ActingAppType,string,Optional,UserManagement,Enumerated,Process|Browser|Service|Resource|URL|SaaS application|Other,
+ActiveAppName,string,Optional,UserManagement,,,
+ActorOriginalUserType,,,UserManagement,,,
+ActorSessionId,string,Optional,UserManagement,,,
+ActorUserId,string,Optional,UserManagement,,,
+ActorUserIdType,string,Optional,UserManagement,Enumerated,SID|UID|AADID|OktaId|AWSId,
+ActorUsername,string,Mandatory,UserManagement,,,
+ActorUsernameType,string,Mandatory,UserManagement,Enumerated,UPN|Windows|DN|Simple,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1131]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 7]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventSchema,string,Mandatory,UserManagement,,,
+EventSchema,string,Mandatory,UserManagement,Enumerated,UserManagement,
+EventSchemaVersion,string,Mandatory,UserManagement,,,
+EventSchemaVersion,string,Mandatory,UserManagement,SchemaVersion,,
+EventSeverity,string,Mandatory,UserManagement,Enumerated,Informational|Low|Medium|High,
+EventSeverity,string,Recommended,UserManagement,Enumerated,,
+EventStartTime,datetime,Mandatory,UserManagement,,,
+EventSubType,string,Optional,UserManagement,Enumerated,,
+EventSubType,string,Optional,UserManagement,Enumerated,UserRead|UserCreated|GroupCreated|UserModified|GroupModified,
+EventType,string,Mandatory,UserManagement,Enumerated,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1177]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 8]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventOriginalResultDetails,string,Optional,UserManagement,,,
+EventOriginalSeverity,string,Optional,UserManagement,,,
+EventOriginalSubType,string,Optional,UserManagement,,,
+EventOriginalType,string,Optional,UserManagement,,,
+EventOriginalUid,string,Optional,UserManagement,,,
+EventOwner,string,Optional,UserManagement,,,
+EventProduct,string,Mandatory,UserManagement,,,
+EventProductVersion,string,Optional,UserManagement,,,
+EventReportUrl,string,Optional,UserManagement,,,
+EventResult,string,Mandatory,UserManagement,Enumerated,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1164]: done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 9]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventResult,string,Mandatory,UserManagement,Enumerated,Success|Partial|Failure|NA,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,,
+EventResultDetails,string,Recommended,UserManagement,Enumerated,NotAuthorized|Other,
+EventSchema,string,Mandatory,UserManagement,,,
+EventSchema,string,Mandatory,UserManagement,Enumerated,UserManagement,
+EventSchemaVersion,string,Mandatory,UserManagement,,,
+EventSchemaVersion,string,Mandatory,UserManagement,SchemaVersion,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1171]: No, we will keep them in line with the others.

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 10]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+EventSeverity,string,Mandatory,UserManagement,Enumerated,Informational|Low|Medium|High,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1172]: Sorted

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 11]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+ActorUsernameType,string,Mandatory,UserManagement,Enumerated,UPN|Windows|DN|Simple,
+ActorUserType,string,Optional,UserManagement,Enumerated,Regular|Machine|Admin|System|Application|Service Principal|Other,
+AdditionalFields,dynamic,Optional,UserManagement,,,
+Dvc,string,Mandatory,UserManagement,,,
+DvcAction,string,Recommended,UserManagement,,,
+DvcDescription,string,Optional,UserManagement,,,
+DvcDomain,string,Recommended,UserManagement,,,
+DvcDomainType,string,Recommended,UserManagement,Enumerated,,
+DvcFQDN,string,Recommended,UserManagement,,,
+DvcHostname,Hostname,Recommended,UserManagement,,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1140]: Yes

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 12]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+GroupNameType,string,Optional,UserManagement,Enumerated,UPN|Windows|DN|Simple,
+GroupOriginalType,string,Optional,UserManagement,,,
+GroupType,string,Optional,UserManagement,Enumerated,Local Distribution|Local Security Enabled|Global Distribution|Global Security Enabled|Universal Distribution|Universal Security Enabled|Other,
+Hostname,string,Alias,UserManagement,,,DvcHostname
+HttpUserAgent,string,Optional,UserManagement,,,
+IpAddr,IP address,Alias,UserManagement,,,SrcIpAddr
+NewPropertyValue,string,Optional,UserManagement,,,
+PreviousPropertyValue,string,Optional,UserManagement,,,
+Src,string,Recommended,UserManagement,,,
+SrcDeviceType,string,Optional,UserManagement,Enumerated,,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1193]: Done

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 13]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+SrcDomain,string,Recommended,UserManagement,,,
+SrcDomainType,string,Recommended,UserManagement,Enumerated,Windows|FQDN,
+SrcDvcId,string,Optional,UserManagement,,,
+SrcDvcIdType,string,Optional,UserManagement,Enumerated,AzureResourceId|MDEid|MD4IoTid|VMConnectionId|AwsVpcId|VectraId|AppGateId|Other,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1197]: combined from other SrcDvcIdType

File: ASIM/schemas/ASimUserManagement.yaml (hunk 1 above)
Comment (KustoKing) [reply ASIM/schemas/ASimUserManagement.yaml:1]: Yes

File: ASIM/dev/ASimTester/ASimTester.csv [hunk 14]
```diff
@@ -1120,6 +1120,99 @@ ActorScope,string,Optional,ProcessEvent,,,
...
+SrcGeoLatitude,Latitude,Optional,UserManagement,,,
+SrcGeoLongitude,Longitude,Optional,UserManagement,,,
+SrcGeoRegion,Region,Optional,UserManagement,,,
+SrcHostname,string,Recommended,UserManagement,,,
+SrcIpAddr,IP address,Recommended,UserManagement,,,
+TargetOriginalUserType,string,Optional,UserManagement,,,
+TargetUserId,string,Optional,UserManagement,,,
+TargetUserIdType,string,Optional,UserManagement,Enumerated,SID|UID|AADID|OktaId|AWSId,
+TargetUsername,string,Optional,UserManagement,,,
+TargetUsernameType,string,Optional,UserManagement,Enumerated,UPN|Windows|DN|Simple,
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply ASIM/dev/ASimTester/ASimTester.csv:1212]: aws is used for userid

**Comment by v-atulyadav** [comment]
Hi @KustoKing,
Thank you for accommodating the suggested changes! We will review the Pull Request internally and get back to you by shortly.

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**